app/controllers/api/v1/classifications_controller.rb

```diff
@@ -36,10 +36,23 @@ def build_resource_for_update(update_params)
     classification
   end
 
+  def sms_ids_from_links(link_params)
+    sms = if ids = link_params.delete(:set_member_subjects)
+      SetMemberSubject.select(:id).find(ids)
```

Comment:
likewise here `SetMemberSubject.where(id: ids).pluck(:id)`
then you wont have to try the array on line 48.